crates/pallet-evm-system/src/migrations/broken_nonces_recovery.rs

```diff
@@ -0,0 +1,189 @@
+//! Migration to recover broken nonces.
+
+#[cfg(feature = "try-runtime")]
+use frame_support::sp_std::vec::Vec;
+use frame_support::{log::info, pallet_prelude::*, traits::OnRuntimeUpgrade};
+use rlp::RlpStream;
+use sp_core::H160;
+use sp_io::hashing::keccak_256;
+use sp_runtime::traits::{CheckedAdd, One, Zero};
+
+use crate::{Account, AccountInfo, Config, Pallet};
+
+/// EVM state provider.
+pub trait EvmStateProvider<AccountId> {
+    /// Check whether account address has code.
+    fn has_code(account_id: &AccountId) -> (bool, Weight);
+
+    /// Gives an iterator over accounts managed by EVM. Considers precompiled contracts' accounts
+    /// as not "managed by EVM".
+    fn accounts_managed_by_evm() -> impl Iterator<Item = (AccountId, Weight)>;
```

Comment:
Just to think, based on conversation that we had about naming related to evm, we remember that each account created in `pallet-evm` already has code. So, what about 
`fn is_managed_by_evm(...)` and `fn all_accounts_managed_by_evm(...)`?